query/src/storages/fuse/operations/read.rs

```diff
@@ -42,14 +42,19 @@ impl FuseTable {
         &self,
         ctx: Arc<QueryContext>,
         push_downs: &Option<Extras>,
+        num_parts: usize,
     ) -> Result<SendableDataBlockStream> {
         let block_reader = self.create_block_reader(&ctx, push_downs)?;
 
-        let bite_size = ctx.get_settings().get_parallel_read_threads()?;
-        let iter = std::iter::from_fn(move || match ctx.clone().try_get_partitions(bite_size) {
-            Err(_) => None,
-            Ok(parts) if parts.is_empty() => None,
-            Ok(parts) => Some(parts),
+        let max_threads = ctx.get_settings().get_max_threads()?;
+        let current_block_reading = num::Integer::div_ceil(&(num_parts as u64), &max_threads);
```

Comment:
All part is concurrent? When `max_ threads = 8` and  `parts.len() = 16`, we will create 8 ✖️ SendableDataBlockStream, and read two parts in each stream?

Reply:
yeah, seems too aggressive & and has no significant effect at least in CPU bounded tasks either.
let me refine it

Reply:
@zhang2014  PTAL